test/generate_parse_fixture_yml.py

```diff
@@ -1,34 +1,39 @@
 """Utility to generate yml files for all the parsing examples."""
+import multiprocessing
 import os
 
 import oyaml as yaml
 
-from sqlfluff.core.parser import Parser, Lexer
-from sqlfluff.core import FluffConfig
-from sqlfluff.cli.commands import quoted_presenter
+from conftest import compute_parse_tree_hash, get_parse_fixtures, parse_example_file
 
-from dialects.parse_fixtures import get_parse_fixtures, load_file
 
-yaml.add_representer(str, quoted_presenter)
-
-parse_success_examples, _ = get_parse_fixtures()
-
-for example in parse_success_examples:
+def generate_parse_fixture(example):
+    """Parse example SQL file, write parse tree to YAML file."""
     dialect, sqlfile = example
-    config = FluffConfig(overrides=dict(dialect=dialect))
-    # Load the SQL
-    raw = load_file(dialect, sqlfile)
-    # Lex and parse the file
-    tokens, _ = Lexer(config=config).lex(raw)
-    tree = Parser(config=config).parse(tokens)
-    r = None
-    if tree:
-        r = tree.as_record(code_only=True, show_raw=True)
+    tree = parse_example_file(dialect, sqlfile)
+    _hash = compute_parse_tree_hash(tree)
     # Remove the .sql file extension
     root = sqlfile[:-4]
     path = os.path.join("test", "fixtures", "parser", dialect, root + ".yml")
     with open(path, "w", newline="\n") as f:
-        if r:
+        r = None
+        if tree:
+            r = dict(
+                [("_hash", _hash)]
+                + list(tree.as_record(code_only=True, show_raw=True).items())
+            )
             yaml.dump(r, f, default_flow_style=False)
```

Comment:
Could we generate a comment line at the top of each YML file as well to really hammer home the point for those that won't run the tests and see the message until after?

For example something like this:
```
# YML test files are auto-generated from SQL files and should not be edited by hand as the hash needs to be calculated.
# Please run the `python test/generate_parse_fixture_yml.py` command to generate them after adding/altering SQL files.
```

Reply:
Will do.

Reply:
That's great. Definitely think that makes this clearer and further avoids people hand editing. Plus points them in the right direction to generate!